### PR TITLE
Massive Ruin Fixes + Removals PR

### PR DESCRIPTION
**Pull request by Zevotech**
## About The Pull Request
This PR is made so I can stop getting angry at the ruins beyond saving that are still ingame. My criteria for a ruin being removed is if another ruin already does its niche better, if the ruin is outdated and/or the ruin is excessively small or unbalanced. For ruins that dont meet this criteria but are still outdated, they will be getting balance fixes and touch ups or a total remap.

This PR is a draft for now because I will need to update the PR changelog and description as I make changes and communicate with the maptainers on what stays and what goes.

Adds departmental RND lootdrop spawners for circuit imprinters, protolathes and techfabs. Excludes omnisci and basic boards from the drops.
Fixed a space tile under a door and replaced the omnilathe with a medical lathe on dangerousresearch
Fixed the whitesands saloon not spawning which may have caused some sandplanets to spawn without a ruin
Fixed harmfactory's nonfunctional traps to now be as lethal as intended. Also changed the loot in the vault to better reflect the ruin's theme and difficulty (cargo techfab board instead of omnilathe, adv plasma cutter instead of combat medkit, less gold more cash, kept the cyberarm implant).
Fixed provinggrounds magical SMES FINALLY by adding a terminal on the back. The map should finally function as intended.
Fixed a few dirs on fire extinguisher cabinets and blast door assemblies in singularity_lab
Removed mechtransport.dmm for being small and bad
Removed some leftover gasthelizards.dmm cruft (VILE)
Removed nucleardump for being an utter mess of an oldcode ruin
Removed gondolaasteroid for being large and empty besides gondolas. better suited for a jungle planet IMO.
Removed Jungle_Spider. Literally just a box with spiders and cloning equipment. Small, bad, hard to find, unjustified loot.
Removed Golem_Hijack. Like jungle spider but it was free rnd, an AI core, a full BSRPED and three golem corpses. With no enemies or obstacles.
Removed rockplanet_clock for being a tiny lootbox that doesnt fit with the lore. Also had a quantumn pad.
Removed whitesands_surface_youreinsane. Its a silly little reference to an old event that unfortunately resulted in a subpar ruin. Could return as a wasteplanet greeble ruin, but it has to go for now.

## Why It's Good For The Game
Normally I'm all for remapping instead of removing ruins, but some ruins are very much beyond saving. Clearing out space for better ruins to take the spotlight is always nice. Some older ruins are fine but are missing certain things or have loot that worked fine in the past, but doesn't reflect the balance we want for ruins in the present.

I will be PR'ing ruins to replace the ones I remove.

## Changelog

:cl:
add: departmental RND lootdrop spawners for imprinters, protolathes and techfabs
fix: dangerous_research.dmm now no longer has a space tile under a door and a medical lathe instead of an omnilathe
fix: whitesands_surface_camp_saloon can now spawn again after its remap into a functional ruin
fix: harmfactory.dmm's traps now work and loot has been adjusted to fit the ruin better
fix: provinggrounds.dmm now has a working SMES and power
fix: singularity_lab fire extinguishers and a few poddoors now have correct dirs
del: mechtransport.dmm and associated code
del: gasthelizards areas
del: nucleardump.dmm and associated code
del: gondolaasteroid.dmm and associated code
del: jungle_spider.dmm and associated code
del: whitesands_golem_hijack.dmm and associated code
del: rockplanet_clock.dmm and associated code
del: whitesands_surface_youreinsane.dmm and associated code
/:cl:

### CODE REVIEW

**Comment by spockye** [comment]
🦐 
its not the shrivel emote, but close enough
i'll go through this tomorrow. if i forget, throw a rock through my window

**Comment by Zevotech** [comment]
> 🦐 its not the shrivel emote, but close enough i'll go through this tomorrow. if i forget, throw a rock through my window

PR's far from done but feel free to drop suggestions in review

**Comment by spockye** [comment]
oh, apologies. missed the Draft. Ping me when you want a proper review

**Comment by Zevotech** [comment]
yeah i guess that'll do. still need more suggestions for ruins to fix or remove by maptainers.

**Comment by Zevotech** [comment]
dont worry im still going to remap the rockplanet_box_robotics, golemcrash_hijack and the other whitesands ruins I was going to get.

**Review by spockye — APPROVED** [review]
sorry for the delay, thanks!

**Comment by Zevotech** [comment]
Actually still have a few more things on my list to add to this but it is in a mergable state. If you have an idea for anything that should go DM me or ping in ruinyard.

**Comment by MarkSuckerberg** [comment]
are you still planning to add more or is this good for merge now (please say yes. new PRs are easy to make)

**Comment by Latentish** [comment]
I don't entirely agree with a couple of the ruins being removed here. You also change the ruin blacklist files, which are deprecated now.
I'm going to DNM this for now while we discuss it.

**Comment by Latentish** [comment]
> I don't entirely agree with a couple of the ruins being removed here. You also change the ruin blacklist files, which are deprecated now. I'm going to DNM this for now while we discuss it.

Hey Latentcy get some glasses.

**Comment by Latentish** [comment]
Remove the blacklist files please. In the meantime we can try TMing this.

**Comment by Zevotech** [comment]
> are you still planning to add more or is this good for merge now (please say yes. new PRs are easy to make)

grins evilly (maybe. I think I've gotten most of the worst offenders that come to mind.)

File: _maps/RandomRuins/SpaceRuins/nuclear_dump.dmm
Comment (Erikafox) [inline _maps/RandomRuins/SpaceRuins/nuclear_dump.dmm:1]: I'm partial to nuclear dump but as long as you have a replacement on the grill.

**Review by Erikafox — APPROVED** [review]

**Pull request merged**